### PR TITLE
fix(devtools): allow ambient modules

### PR DESCRIPTION
**Pull request by daniellacosse**
Allows triple-slash directive in our eslint. Also typechecks the png import.

### CODE REVIEW

File: src/www/web-dev-server.config.mjs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {esbuildPlugin} from '@web/dev-server-esbuild';
```
Comment (daniellacosse) [inline src/www/web-dev-server.config.mjs:1]: I moved this into `src/www` so it's next to the storybook it launches.

File: src/www/tsconfig.json [hunk 2]
```diff
@@ -1,9 +1,7 @@
 {
-  "extends": "../tsconfig.json",
+  "extends": "../../tsconfig.json",
   "compilerOptions": {
     "outDir": "../../www"
   },
-  "include": [
-    "**/*.ts"
-  ]
+  "exclude": ["../../www/**/*"]
```
Comment (daniellacosse) [inline src/www/tsconfig.json:6]: It's intermittent, but without these excludes you get this weird error in VSCode saying it's trying to overwrite its own input files.

File: tsconfig.json [hunk 3]
```diff
@@ -5,11 +5,13 @@
...
     "noImplicitThis": true,
-    "moduleResolution": "Node",
+    "moduleResolution": "node",
     "module": "commonjs",
     "resolveJsonModule": true,
     "esModuleInterop": true,
     "allowJs": true,
-    "typeRoots": ["../node_modules/@types", "./types/modules"]
-  }
+    "skipLibCheck": true
```
Comment (daniellacosse) [inline tsconfig.json:13]: This limits TS to check only what's explicitly in our source code, and no external libraries

Comment (daniellacosse) [reply tsconfig.json:13]: "typeRoots" was constraining the types pulled in to the TS compilation process such that we had to explicitly import them with the triple-slash directives

File: src/www/tsconfig.json (hunk 2 above)
Comment (daniellacosse) [reply src/www/tsconfig.json]: (moved to the root tsconfig as just `www`)

File: src/www/app/clipboard.ts [hunk 4]
```diff
@@ -12,8 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/// <reference path='../../types/ambient/clipboard.d.ts'/>
```
Comment (fortuna) [inline src/www/app/clipboard.ts:15]: These directives is the only way to express that this file depends on that type.
Why are they not allowed? What's the suggested alternative?
I'd much rather keep a way to represent the dependency.

File: tsconfig.json [hunk 5]
```diff
@@ -5,11 +5,13 @@
...
+  },
+  "include": ["./src/**/*.ts"],
```
Comment (fortuna) [inline tsconfig.json:15]: I think this file is only used as a base for the other tsconfig files, right? Why does it need to specify include? Shouldn't that be in the other files? Isn't that defined by webpack? It feels redundant or unused.

File: .eslintrc.json [hunk 6]
```diff
@@ -32,6 +32,14 @@
       "rules": {
         "@typescript-eslint/no-explicit-any": "off"
       }
+    },
+    {
+      "files": ["*.js", "*.mjs"],
+      "extends": ["eslint:recommended"],
+      "rules": {
+        "@typescript-eslint/no-explicit-any": "off",
```
Comment (fortuna) [inline .eslintrc.json:40]: Is this so we don't need to fix them in this PR or is this the recommendation?
Perhaps add a comment?

File: src/www/app/clipboard.ts (hunk 4 above)
Comment (daniellacosse) [reply src/www/app/clipboard.ts:15]: I forgot you could just use "import" so i did that instead, even though it's now unnecessary (they are implicitly imported everywhere)

File: tsconfig.json (hunk 5 above)
Comment (daniellacosse) [reply tsconfig.json:16]: Added a comment.

File: .eslintrc.json (hunk 6 above)
Comment (daniellacosse) [reply .eslintrc.json:41]: Added comment about intent.

**Pull request merged**